code/src/reducer/quests.js

```diff
@@ -0,0 +1,35 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const questsList = []
+const categories = ['Main', 'Side', 'Miscellaneous']
+
+const initialState = {
+  questsList,
+  categories
+}
+
+export const quests = createSlice({
+  name: 'quests',
+  initialState,
+  reducers: {
+    createQuest: (state, action) => {
+      const { text, category } = action.payload
+      state.questsList.push({
```

Comment:
Don't forget the importance of IDs, or some form of unique keys in react. In your code, if you were to implement reordering of quests, react would have a hard time to see something has changed since you're relying entirely of the index of items in the array. It's much more solid to use unique IDs to track things since the array order can change